src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -95,4 +95,16 @@
      * @return True if can be interrupted, false if not
      */
     boolean canBeInterrupted();
+
+    /**
+     * Resets this task.
+     */
+    void reset();
+
+    /**
+     * Gets the mutex bits. When the | operation result of two mutex bits is 0, they can run
+     * concurrently.
+     * @return The mutex bits which categorizes this task
+     */
+    int getMutexBits();
```

Comment:
The issue with canRunConcurrentWith is that you have to know details about every task/goal. This is impossible for plugin based tasks and thus a definition of affected elements from the entity movement/view etc are important to know to check whether the actualy can concurrently.

If you dont want them everywhere then you could still move it to a seperate interface.